mathoxide-lib/src/array.rs

```diff
@@ -72,6 +73,27 @@ where
         let storage = StorageType::from(v);
         Array { storage, view }
     }
+
+    pub fn reshape<SizeType, ListType>(&self, shape: ListType) -> Self
+    where
+        SizeType: Copy,
+        ShapeDim: From<SizeType>,
+        ListType: AsRef<[SizeType]>,
+    {
+        let new_shape = shape_utils::infer_shape(shape, self.numel());
```

Comment:
why not `use` the `infer_shape` directly?

Reply:
what do you think about having a struct `ShapeUtils` that has infer_shape as a method? Then we do something like `ShapeUtils::new(&self.view).infer_shape(shape)`. This will be more consistent with the Formatter. Alternatively, we could have `infer_shape` or even a `reshape` inside the view?